resources/aws_ec2.go

```diff
@@ -708,3 +718,30 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 		log.Printf(string(post))
 	}
 }
+
+// snsMakeTopic creates a topic on aws sns.
+func (obj *AwsEc2Res) snsMakeTopic() error {
+	// make sns client
+	snsSess, err := session.NewSession(&aws.Config{
+		Region: aws.String(obj.Region),
+	})
+	if err != nil {
+		return err
+	}
+	obj.snsClient = sns.New(snsSess)
+
+	// make topic
+	topicInput := &sns.CreateTopicInput{
+		Name: aws.String(topicName),
+	}
+	topic, err := obj.snsClient.CreateTopic(topicInput)
+	if err != nil {
+		return err
+	}
+	log.Printf("%s: Created SNS Topic", obj)
+
+	if topic.TopicArn != nil {
+		obj.topicArn = *topic.TopicArn
```

Comment:
why do we store this instead of returning it?

Reply:
This will be used in snsSubscribe, snsAuthorize, cweTargetRule, and subscribe. I suppose I could return it and pass it in as an argument to those functions if you think that's cleaner. It is also used in Close() which means I'd have to get the topicArn again another way. There's sns.ListTopics, but that returns all topics, not just the one we're concerned with.

Reply:
I'm not really sure. But it might be worth rebasing the large PR with everything in it so I can get a slimmed down view of the big picture again.